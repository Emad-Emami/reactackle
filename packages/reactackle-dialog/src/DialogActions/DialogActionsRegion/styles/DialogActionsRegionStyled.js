'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  alignLeft: PropTypes.bool,
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  alignLeft: false,
};

/** Prop Receivers */
const align = ({ alignLeft }) =>
  alignLeft
    ? `
    text-align: left;
    justify-content: flex-start;
  `
    : `
    text-align: right;
    justify-content: flex-end;
  `;

const childrenSpacing = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { marginX, marginY } = theme.reactackle.components.dialog.actionItem;

  return `
    & > * {
      margin: ${getValueString(marginY)} ${getValueString(marginX)};
    }
  `;
};

/** Styles */
export const DialogActionsRegionStyled = styled.div`
  flex-grow: 1;
  display: flex;
  margin: 0 !important;
  ${align} ${childrenSpacing};
`;

DialogActionsRegionStyled.propTypes = propTypes;
DialogActionsRegionStyled.defaultProps = defaultProps;
DialogActionsRegionStyled.displayName = 'DialogActionsRegionStyled';
