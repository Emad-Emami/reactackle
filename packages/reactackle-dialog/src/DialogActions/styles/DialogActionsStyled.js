'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const baseProps = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { paddingX, paddingY } = theme.reactackle.components.dialog.actions;

  const { marginX, marginY } = theme.reactackle.components.dialog.actionItem;

  const py =
    typeof paddingY === 'number' && typeof marginY === 'number'
      ? `${paddingY - marginY}px`
      : `calc(${getValueString(paddingY)} - ${getValueString(marginY)})`;

  const px =
    typeof paddingX === 'number' && typeof marginX === 'number'
      ? `${paddingX - marginX}px`
      : `calc(${getValueString(paddingX)} - ${getValueString(marginX)})`;

  return `
    padding: ${py} ${px};

    > * {
      margin: ${getValueString(marginX)} ${getValueString(marginY)};
    }
  `;
};

export const DialogActionsStyled = styled.div`
  position: relative;
  display: flex;
  justify-content: flex-end;
  flex-shrink: 0;
  ${baseProps};
`;

DialogActionsStyled.propTypes = propTypes;
DialogActionsStyled.displayName = 'DialogActionsStyled';
