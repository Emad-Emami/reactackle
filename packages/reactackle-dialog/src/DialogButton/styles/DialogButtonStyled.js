'use strict';

import styled from 'styled-components';

export const DialogButtonStyled = styled.div`
  display: flex;
  flex-shrink: 0;
`;

DialogButtonStyled.displayName = 'DialogButtonStyled';
