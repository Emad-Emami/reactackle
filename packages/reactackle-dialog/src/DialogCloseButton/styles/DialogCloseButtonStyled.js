'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
  transition,
} from 'reactackle-core';

const propTypes = {
  transparentBg: PropTypes.bool,
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  transparentBg: false,
};

/** Prop Receivers */
const sizeProps = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    width,
    height,
    imgSize,
  } = theme.reactackle.components.dialog.closeButton;

  return `
    width: ${getValueString(width)};
    height: ${getValueString(height)};
    line-height: ${getValueString(height)};
    font-size: ${getValueString(imgSize)};
  `;
};

const styleProps = ({ theme: themeFromProvider, transparentBg }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const path = theme.reactackle.components.dialog.closeButton.style,
    source = transparentBg ? path.modalWithTransparentBg : path;

  return `
    color: ${source.color};
    opacity: ${source.opacity};

    &:hover {
      color: ${source.hover.color};
      opacity: ${source.hover.opacity};
    }

    &:focus {
      color: ${source.focus.color};
      opacity: ${source.focus.opacity};
    }
  `;
};

/** Styles */
export const DialogCloseButtonStyled = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  display: flex;
  z-index: 9;
  cursor: pointer;
  ${sizeProps} ${styleProps} ${transition('color, opacity')};
`;

DialogCloseButtonStyled.propTypes = propTypes;
DialogCloseButtonStyled.defaultProps = defaultProps;
DialogCloseButtonStyled.displayName = 'DialogCloseButtonStyled';
