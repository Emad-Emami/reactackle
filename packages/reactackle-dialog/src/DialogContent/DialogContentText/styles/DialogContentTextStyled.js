'use strict';

import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

/** Prop Receivers */
const baseProps = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    fontSize,
    lineHeight,
    color,
    fontWeight,
    similarTextBlockSpacing,
  } = theme.reactackle.components.dialog.content;

  return css`
    fontSize: ${getValueString(fontSize)};
    line-height: ${lineHeight};
    color: ${color};
    font-weight: ${fontWeight};
    
    & + & {
      margin-top: ${getValueString(similarTextBlockSpacing)};
    }
  `;
};

/** Styles */
export const DialogContentTextStyled = styled.div`
  margin: 0;
  ${baseProps};
`;

DialogContentTextStyled.propTypes = propTypes;
DialogContentTextStyled.displayName = 'DialogContentTextStyled';
