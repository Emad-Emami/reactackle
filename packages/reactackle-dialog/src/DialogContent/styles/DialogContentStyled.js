'use strict';

import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  bordered: PropTypes.bool,
  scrollable: PropTypes.bool,
  displayFlex: PropTypes.bool,
  followsHeading: PropTypes.bool,
  paddingSize: PropTypes.oneOf(['none', 'default']),
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  bordered: false,
  scrollable: false,
  displayFlex: false,
  followsHeading: false,
  paddingSize: 'default',
};

/** Prop Receivers */
const baseProps = ({
  theme: themeFromProvider,
  paddingSize,
  followsHeading,
  bordered,
  displayFlex,
}) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    paddingX,
    paddingY,
    fontSize,
    lineHeight,
    color,
    fontWeight,
    borderWidth,
    borderColor,
  } = theme.reactackle.components.dialog.content;

  const py = getValueString(paddingY),
    px = getValueString(paddingX);

  const padding = paddingSize !== 'none' ? `padding: ${py} ${px};` : '';

  return css`
    font-size: ${getValueString(fontSize)};
    line-height: ${lineHeight};
    color: ${color};
    font-weight: ${fontWeight};

    ${padding}

    ${followsHeading && 'padding-top: 0;'}

    & + & {
      padding-top: 0;
      margin-top: -calc(${py} / 2);
    }

    ${bordered
      ? `border-top: ${getValueString(borderWidth)} solid ${borderColor};`
      : ''}
    
    ${displayFlex && 'display: flex;'}
  `;
};

const scrollable = ({ scrollable }) => (scrollable ? 'overflow-y: auto;' : '');

/** Styles */
export const DialogContentStyled = styled.div`
  flex-grow: 1;
  position: relative;
  ${baseProps} ${scrollable};
`;

DialogContentStyled.propTypes = propTypes;
DialogContentStyled.defaultProps = defaultProps;
DialogContentStyled.displayName = 'DialogContentStyled';
