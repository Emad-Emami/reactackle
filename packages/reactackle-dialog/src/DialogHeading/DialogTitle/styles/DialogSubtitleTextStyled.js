'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';

import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const propTypes = {
  /**
   * Define theme
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

/** Prop Receivers */
const baseProps = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    fontSize,
    lineHeight,
    color,
    fontWeight,
    subtitleTitleSpacing,
  } = theme.reactackle.components.dialog.subtitle;

  return `
    font-size: ${getValueString(fontSize)};
    line-height: ${lineHeight};
    color: ${color};
    font-weight: ${fontWeight};
    margin-top: ${getValueString(subtitleTitleSpacing)};
  `;
};

/** Styles */
export const DialogSubtitleTextStyled = styled.div`
  margin: 0;
  ${baseProps};
`;

DialogSubtitleTextStyled.propTypes = propTypes;
DialogSubtitleTextStyled.displayName = 'DialogSubtitleTextStyled';
