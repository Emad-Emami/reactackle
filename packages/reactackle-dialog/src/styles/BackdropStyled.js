'use strict';

import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  extractThemeOrDefault,
  getValueString,
  animations,
} from 'reactackle-core';

const propTypes = {
  theme: PropTypes.object,
};

const baseProps = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { margin } = theme.reactackle.components.dialog.window;

  const {
    backgroundColor,
    zIndex,
  } = theme.reactackle.components.dialog.backdrop;

  return `
    padding: ${getValueString(margin)};
    z-index: ${zIndex};
    background-color: ${backgroundColor};
    animation: ${animations.fadeIn} 300ms ease-in;
  `;
};

export const BackdropStyled = styled.div`
  display: block;
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  ${baseProps};
`;

BackdropStyled.propTypes = propTypes;
BackdropStyled.displayName = 'BackdropStyled';
