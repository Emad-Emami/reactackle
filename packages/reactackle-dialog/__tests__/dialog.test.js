'use strict';

import React from 'react';
import {
  findByType,
  rendererWithPortal as rendererWithPortalMain,
 } from 'reactackle-test-utils';
import Portal from 'react-portal';

import {
  Dialog,
  DialogContentText,
} from '../src';

import { BackdropStyled } from '../src/styles/BackdropStyled';

const rendererWithPortal = (component, options) =>
  rendererWithPortalMain(component, options, Portal);
jest.mock('react-dom');

const map = {};

document.body.addEventListener = jest.fn((event, cb) => {
  map[event] = cb;
});

describe('<Dialog/>', () => {
  it('renders correctly with default props', () => {
    const component = rendererWithPortal(
      <Dialog />,
    );
    expect(component.toJSON()).toMatchSnapshot();
    component.unmount();
  });

  it('renders correctly if contains children', () => {
    const component = rendererWithPortal(
      <Dialog open>
        <div> test </div>
      </Dialog>,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly call button onPress if contains buttons', () => {
    const mockFn = jest.fn(),
      buttons = [{
        text: 'Button 1',
        onPress: mockFn,
      }],
      component = rendererWithPortal(
        <Dialog open buttons={buttons}>
          <div> test </div>
        </Dialog>,
      );
    
    findByType(component.toJSON(), 'button').props.onClick();

    expect(component.toJSON()).toMatchSnapshot();
    expect(mockFn).toHaveBeenCalledTimes(1);
  });


  it('renders correctly if prop backdrop set true', () => {
    const component = rendererWithPortal(
      <Dialog backdrop open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly if prop backdrop set and click on backdrop', () => {
    const component = rendererWithPortal(
      <Dialog backdrop open closeOnBackdropClick />,
    );

    expect(component.toJSON()).toMatchSnapshot();

    findByType(component, BackdropStyled).props.onClick();

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop scrollable set true', () => {
    const component = rendererWithPortal(
      <Dialog scrollable open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop open set true', () => {
    const component = rendererWithPortal(
      <Dialog open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop haveCloseButton set true', () => {
    const component = rendererWithPortal(
      <Dialog haveCloseButton open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop closeOnEscape set true', () => {
    const component = rendererWithPortal(
      <Dialog closeOnEscape open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop closeOnBackdropClick set true', () => {
    const component = rendererWithPortal(
      <Dialog closeOnBackdropClick open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop title set title', () => {
    const component = rendererWithPortal(
      <Dialog title="title" open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop subtitle set subtitle', () => {
    const component = rendererWithPortal(
      <Dialog subtitle="subtitle" open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if prop minWidth > 0', () => {
    const component = rendererWithPortal(
      <Dialog minWidth={50} open />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if has prop regionAside', () => {
    const component = rendererWithPortal(
      <Dialog open regionAside={<div>RegionAside</div>} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  
  it('renders with DialogContentText', () => {
    const component = rendererWithPortal(
      <Dialog open>
        <DialogContentText>
          Texty text
        </DialogContentText>
      </Dialog>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  
  it('renders with windowTitle', () => {
    const component = rendererWithPortal(
      <Dialog open windowTitle="Window title" />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
    
  it('handles keyup and props change correctly if open', () => {
    const component = rendererWithPortal(
      <Dialog windowTitle="Window title" backdrop open closeOnEscape />,
    );

    map.keyup({ keyCode: 27 });
    map.keyup({ keyCode: 13 });

    expect(component.toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly left buttons', () => {
    const buttons = [{
        text: 'Button',
      }],
      component = rendererWithPortal(
        <Dialog open buttonsLeft={buttons}>
          <div> test </div>
        </Dialog>,
      );
    
    findByType(component.toJSON(), 'button').props.onClick();
    expect(component.toJSON()).toMatchSnapshot();
  });
});
