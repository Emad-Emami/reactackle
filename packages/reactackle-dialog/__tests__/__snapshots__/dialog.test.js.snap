// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog/> handles keyup and props change correctly if open 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="BackdropStyled__BackdropStyled-htftey dVajva"
    onClick={[Function]}
  />
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly call button onPress if contains buttons 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        >
          <div>
             test 
          </div>
        </div>
        <div
          className="DialogActionsStyled__DialogActionsStyled-kfIYNh enfxMJ"
        >
          <div
            className="DialogActionsRegionStyled__DialogActionsRegionStyled-fBBQFS ceLbqn"
          >
            <div
              className="DialogButtonStyled__DialogButtonStyled-GXazH gcUwPO"
            >
              <button
                className="ButtonStyled__Button-FHerW fLKthV"
                disabled={false}
                onClick={[Function]}
                radius="default"
                size="normal"
                title=""
              >
                <div
                  className="ButtonTextStyled__ButtonTextStyled-dokorR eDbaCA"
                  size="normal"
                >
                  <div
                    className="ButtonTitleStyled__ButtonTitleStyled-kwhLfn koUihP"
                    size="normal"
                  >
                    Button 1
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if contains children 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        >
          <div>
             test 
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if has prop regionAside 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
      <div
        className="RegionSideStyled__RegionSideStyled-edRfvg erohFc"
      >
        <div>
          RegionAside
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop backdrop set and click on backdrop 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="BackdropStyled__BackdropStyled-htftey dVajva"
    onClick={[Function]}
  />
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop backdrop set and click on backdrop 2`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="BackdropStyled__BackdropStyled-htftey dVajva"
    onClick={[Function]}
  />
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop backdrop set true 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="BackdropStyled__BackdropStyled-htftey dVajva"
    onClick={[Function]}
  />
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop closeOnBackdropClick set true 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop closeOnEscape set true 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop haveCloseButton set true 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogCloseButtonStyled__DialogCloseButtonStyled-hWZfNE jMBkHu"
          onClick={[Function]}
        >
          <span
            className="fa fa-times IconFontAwesomeStyled__IconFontAwesomeStyled-cpbEq eVGVUG"
            color=""
            onClick={[Function]}
            rotate={0}
            size="normal"
          />
        </div>
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop minWidth > 0 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB kfdCjZ"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop open set true 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop scrollable set true 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd eDDkUq"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop subtitle set subtitle 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogHeadingStyled__DialogHeadingStyled-iSBdOK dFdOom"
        >
          <div
            className="DialogTitleStyled__DialogTitleStyled-gGKTRE eKXiYf"
          >
            <div
              className="DialogTitleTextStyled__DialogTitleTextStyled-bLMOnG kvtbwo"
            >
              
            </div>
            <div
              className="DialogSubtitleTextStyled__DialogSubtitleTextStyled-iBJFax hBUibh"
            >
              subtitle
            </div>
          </div>
        </div>
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iuWqkD"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly if prop title set title 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogHeadingStyled__DialogHeadingStyled-iSBdOK dFdOom"
        >
          <div
            className="DialogTitleStyled__DialogTitleStyled-gGKTRE eKXiYf"
          >
            <div
              className="DialogTitleTextStyled__DialogTitleTextStyled-bLMOnG kvtbwo"
            >
              title
            </div>
          </div>
        </div>
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iuWqkD"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly left buttons 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        >
          <div>
             test 
          </div>
        </div>
        <div
          className="DialogActionsStyled__DialogActionsStyled-kfIYNh enfxMJ"
        >
          <div
            className="DialogActionsRegionStyled__DialogActionsRegionStyled-fBBQFS MOSGq"
          >
            <div
              className="DialogButtonStyled__DialogButtonStyled-GXazH gcUwPO"
            >
              <button
                className="ButtonStyled__Button-FHerW fLKthV"
                disabled={false}
                onClick={[Function]}
                radius="default"
                size="normal"
                title=""
              >
                <div
                  className="ButtonTextStyled__ButtonTextStyled-dokorR eDbaCA"
                  size="normal"
                >
                  <div
                    className="ButtonTitleStyled__ButtonTitleStyled-kwhLfn koUihP"
                    size="normal"
                  >
                    Button
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders correctly with default props 1`] = `null`;

exports[`<Dialog/> renders with DialogContentText 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        >
          <div
            className="DialogContentTextStyled__DialogContentTextStyled-izazft hfAHog"
          >
            Texty text
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Dialog/> renders with windowTitle 1`] = `
<div
  className="ModalStyled__ModalStyled-blxqLg hyyLBo"
>
  <div
    className="OverflowWrapperStyled__OverflowWrapperStyled-jecKbB bjjqzT"
  >
    <div
      className="ContentWrapperStyled__ContentWrapperStyled-idYDJV icblmW"
    >
      <div
        className="RegionMainStyled__RegionMainStyled-dpVyvY cgaytx"
      >
        <div
          className="DialogContentStyled__DialogContentStyled-iDwqMd iLDSDl"
        />
      </div>
    </div>
  </div>
</div>
`;
